### PR TITLE
Added canvas context menu and duplicate layer command

### PR DESCRIPTION
**Pull request by CPKreuz**
Added a canvas context menu, containing (De)select all, Copy, Cut, Paste, Bake Pizza and added a duplicate layer command to the layer list

### CODE REVIEW

**Review by flabbet — CHANGES_REQUESTED** [review]
Insert (1) at the end of the duplicated layer name, add undo support. I don't think you need to write any undo process, you can reuse what already exists

File: PixiEditor/Models/DataHolders/Document/Document.Layers.cs [hunk 1]
```diff
@@ -437,6 +469,65 @@ private void RemoveLayerProcess(object[] parameters)
...
+                        if (highestValue == null)
+                        {
+                            highestValue = 0;
+                        }
+                    }
+                }
+            }
+
+            return highestValue;
+        }
```
Comment (flabbet) [inline PixiEditor/Models/DataHolders/Document/Document.Layers.cs:529]: This is a bit too long function, split it a little, please :)

**Review by flabbet — CHANGES_REQUESTED** [review]
The code is clean, everything works, I was unable to break it, you've reused the code well and even wrote a test for suffixes, perfect. The only thing I need to ask you is to write a test for the Duplicate layer. Nothing fancy, just if it's added to Layers and is active, you don't need to compare contents or test undo.

**Review by flabbet — APPROVED** [review]
👌
